pkg/runtime/reconciler.go

```diff
@@ -325,6 +325,17 @@ func (r *resourceReconciler) cleanup(
 		return err
 	}
 	if err = rm.Delete(ctx, observed); err != nil {
+		var requeueNeededAfter *requeue.RequeueNeededAfter
+		var requeueNeeded *requeue.RequeueNeeded
+		if errors.As(err, &requeueNeededAfter) ||
+			errors.As(err, &requeueNeeded) {
+			// read delete progress and corresponding conditions
+			// and patch resource to surface the delete progress.
+			observed, _ := rm.ReadOne(ctx, current)
+			if observed != nil {
+				_ = r.patchResource(ctx, current, observed)
+			}
```

Comment:
Why ignore the returned errors here?

Reply:
This block of code executes when Delete api returned error and requeue is requested. ReadOne error is ignored as during the next reconciler loop run (due to requeue) the correct course of action would be taken.
Error from path resources are ignored throughout in the reconciler at other places as well, thus followed the same.

Reply:
> This block of code executes when Delete api returned error and requeue is requested. ReadOne error is ignored as during the next reconciler loop run (due to requeue) the correct course of action would be taken.
> Error from path resources are ignored throughout in the reconciler at other places as well, thus followed the same.

@kumargauravsharma I think @A-Hilaly was referring to ignoring the returned error from `rm.ReadOne()`, not patchResource... (I think?) :)

Reply:
I thought the comment was for both. 
For error during read one: 
  There are two errors available at that time a) error returned from Delete() method b) error returned from Delete call.
  The code returns the error received from Delete call as that is the primary operation being done in this method.
  ReadOne error is ignored as during the next reconciler loop run (due to requeue) the correct course of action would be taken.

Reply:
IMO we should at least log these errors. Tt would be hard to track them if the metrics server is reporting few 4XX/5XX but logs are not showing any

Reply:
> IMO we should at least log these errors. It would be hard to track them if the metrics server is reporting few 4XX/5XX but logs are not showing any

nvm this was added in the recent logging PR :)